### ISSUE TITLE
`worker` could be `undefined` after `await getClient()`

### ISSUE THREAD

**Issue by JounQin** [remcohaszing/monaco-yaml#163]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `monaco-yaml@4.0.0-alpha.1` for the project I'm working on.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/monaco-yaml/index.js b/node_modules/monaco-yaml/index.js
index d89e293..c0a34a1 100644
--- a/node_modules/monaco-yaml/index.js
+++ b/node_modules/monaco-yaml/index.js
@@ -398,7 +398,10 @@ function createWorkerManager(defaults) {
   };
   return async (...resources) => {
     const client2 = await getClient();
-    await worker.withSyncedResources(resources);
+    // could be cleaned by stopWorker
+    if (worker) {
+      await worker.withSyncedResources(resources);
+    }
     return client2;
   };
 }
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by remcohaszing** [comment]
Thanks for reporting this!

#160 is an open pull request that extracts this logic into a reusable package.

As can be seen in https://github.com/remcohaszing/monaco-worker-manager/blob/main/index.ts#L120-L129, the worker is now assigned synchronously before this line is called, so I think this will be fixed purely by coincidence in the next release.

**Comment by JounQin** [comment]
OK, I'll use my patch temporarily.

When will https://github.com/remcohaszing/monaco-yaml/pull/160 be merged?

**Comment by remcohaszing** [comment]
I tend to wait for someone to do a code review, as @fleon and @yazaabed have both shown interested in helping out. I really appreciate this, but sometimes it takes quite a long time for them to perform code reviews.

If you really need this fix, I can just merge and release it without a review this weekend.

There are some pending new features too. :)

**Comment by remcohaszing** [comment]
I just released [v4.0.0-alpha.2](https://github.com/remcohaszing/monaco-yaml/releases/tag/v4.0.0-alpha.2)
